src/Scan/RestScans.spec.ts

```diff
@@ -0,0 +1,130 @@
+import { RequestPromiseAPI } from 'request-promise';
+import nock from 'nock';
+
+describe('RestScans', () => {
+  let restScans!: RestScans;
+  const moduleMock = mock<Module>();
+  const cliInfoMock = mock<CliInfo>();
+  const testTypeMock = mock<TestType>();
+  const requestPromiseAPIMock = mock<RequestPromiseAPI>();
+
+  beforeAll(() => {
+    nock.disableNetConnect();
+    nock.enableNetConnect('127.0.0.1');
+  });
+
+  beforeEach(() => {
+    if (!nock.isActive()) {
+      nock.activate();
+    }
+
+    restScans = new RestScans(instance(cliInfoMock), {
+      baseUrl: 'https://development.playground.neuralegion.com',
+      apiKey: 'key'
+    });
+  });
+
+  afterEach(() => {
+    reset<CliInfo | TestType | RequestPromiseAPI | Module>(
+      cliInfoMock,
+      testTypeMock,
+      requestPromiseAPIMock,
+      moduleMock
+    );
+    nock.cleanAll();
+    nock.restore();
+  });
+
+  afterAll(() => nock.enableNetConnect());
+
+  describe('create', () => {
+    it('should create a new scan if the HAR file passed by id exists', async () => {
+      // arrange
+      const postResponce = { id: 'string' };
+      const scanConfig: ScanConfig = {
+        name: 'scan',
+        module: instance(moduleMock),
+        tests: [instance(testTypeMock)],
+        fileId: 'id'
+      };
+      const file: StorageFile = { id: scanConfig.fileId, type: SourceType.HAR };
+
+      nock('https://development.playground.neuralegion.com')
+        .replyContentLength()
+        .get(`/api/v1/files/${scanConfig.fileId}`)
+        .reply(200, file, {
+          'content-type': 'application/json'
+        });
+
+      nock('https://development.playground.neuralegion.com')
+        .replyContentLength()
+        .post('/api/v1/scans')
+        .reply(200, postResponce, {
+          'content-type': 'application/json'
+        });
+
+      // act
+      const result = await restScans.create(scanConfig);
+
+      // assert
+      expect(result).toEqual(postResponce.id);
+    });
+
+    it('should create a new scan if the OAS file passed by id exists', async () => {
+      // arrange
+      const postResponce = { id: 'string' };
+      const scanConfig: ScanConfig = {
+        name: 'scan',
+        module: instance(moduleMock),
+        tests: [instance(testTypeMock)],
+        fileId: 'id'
+      };
+      const file: StorageFile = {
+        id: scanConfig.fileId,
+        type: SourceType.OPEN_API
+      };
+
+      nock('https://development.playground.neuralegion.com')
+        .replyContentLength()
+        .get(`/api/v1/files/${scanConfig.fileId}`)
+        .reply(200, file, {
+          'content-type': 'application/json'
+        });
+
+      nock('https://development.playground.neuralegion.com')
+        .replyContentLength()
+        .post('/api/v1/scans')
+        .reply(200, postResponce, {
+          'content-type': 'application/json'
+        });
+
+      // act
+      const result = await restScans.create(scanConfig);
+
+      // assert
```

Comment:
You should also check that `discoveryType` is correctly parsed and passed to `/api/v1/scans`
Here's a workaround for nock on how to get request body
https://github.com/nock/nock/issues/214#issuecomment-395262607
Then you'll need to do something like `.toMatchObject({ discoveryType: expect.arrayContaining(...) })`